core/driver/sources/src/main/scala/org/virtuslab/ideprobe/dependencies/IntelliJResolver.scala

```diff
@@ -37,9 +43,44 @@ case class IntelliJPatternResolver(pattern: String) extends IntelliJResolver {
       "version" -> version.releaseOrBuild,
       "release" -> version.release.getOrElse("snapshot-release")
     )
-    val replaced = replacements.foldLeft(pattern) { case (path, (pattern, replacement)) =>
+    val replacedBeforeResolvingGlobs = replacements.foldLeft(pattern) { case (path, (pattern, replacement)) =>
       path.replace(s"[$pattern]", replacement)
     }
+    val replaced = replaceGlobsWithExistingDirectories(List(replacedBeforeResolvingGlobs)).head
```

Comment:
shouldn't there be a function that accepts one argument and returns one value

Reply:
Hm, I don't know if it would be possible. `replaceGlobsWithExistingDirectories` is recursive and returns multiple paths as a result - by replacing wildcards.

Reply:
One thing I can do is to wrap `replaceGlobsWithExistingDirectories` in another helper function that accepts one argument and returns one value. But the  `replaceGlobsWithExistingDirectories`'s argument- and return- types must stay the same.

Reply:
Yes, please do. It is also quite common practice for such helper recursive function to be nested inside the actual function with expected api (to hide implementation details of recurrence).